### PR TITLE
Support async components

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
* In one the last commits we have change the import for some components from the static [import statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) to the [dynamic import](https://webpack.js.org/guides/code-splitting/#dynamic-imports) function. This has been done because bundles [Async-Components](https://vuejs.org/v2/guide/components-dynamic-async.html#Async-Components) into separate files which are reuses across pages. The problem is that `$refs` to dynamically imported components are not yet defined in the mounted hooks of the parent component. This PR fixes this problem.

* In this PR we also remove all the unused SourceClear yaml file.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement user

```

### CODE REVIEW

File: frontend/src/components/PurposeConfiguration.vue [hunk 1]
```diff
@@ -113,7 +117,7 @@ export default {
     },
     reset () {
       this.purpose = this.shootPurpose
-      this.$refs.purpose.setPurpose(this.purpose)
+      return this.$purpose.dispatch('setPurpose', this.purpose)
```
Comment (holgerkoser) [inline frontend/src/components/PurposeConfiguration.vue:120]: check usage

File: frontend/src/components/NewShoot/NewShootDetails.vue [hunk 2]
```diff
@@ -241,14 +247,14 @@ export default {
         purpose: this.purpose
       }
     },
-    setDetailsData ({ name, kubernetesVersion, purpose, cloudProfileName, secret, updateK8sMaintenance }) {
+    async setDetailsData ({ name, kubernetesVersion, purpose, cloudProfileName, secret, updateK8sMaintenance }) {
```
Comment (holgerkoser) [inline frontend/src/components/NewShoot/NewShootDetails.vue:250]: check usage

File: frontend/src/components/NewShoot/NewShootDetails.vue [hunk 3]
```diff
@@ -241,14 +247,14 @@ export default {
...
       this.name = name
       this.cloudProfileName = cloudProfileName
       this.secret = secret
       this.kubernetesVersion = kubernetesVersion
       this.updateK8sMaintenance = updateK8sMaintenance
 
-      this.$refs.purpose.setPurpose(purpose)
+      await this.$purpose.dispatch('setPurpose', purpose)
```
Comment (holgerkoser) [inline frontend/src/components/NewShoot/NewShootDetails.vue:257]: should not be awaited

File: frontend/src/components/ShootHibernation/HibernationConfiguration.vue [hunk 4]
```diff
@@ -23,10 +23,12 @@ limitations under the License.
     caption="Configure Hibernation Schedule">
     <template v-slot:actionComponent>
       <manage-hibernation-schedule
-        ref="hibernationSchedule"
+        id="hibernationSchedule"
```
Comment (holgerkoser) [inline frontend/src/components/ShootHibernation/HibernationConfiguration.vue:26]: remove

File: frontend/src/components/PurposeConfiguration.vue (hunk 1 above)
Comment (holgerkoser) [reply frontend/src/components/PurposeConfiguration.vue]: ✔️

File: frontend/src/components/NewShoot/NewShootDetails.vue (hunk 2 above)
Comment (holgerkoser) [reply frontend/src/components/NewShoot/NewShootDetails.vue:250]: ✔️

File: frontend/src/components/NewShoot/NewShootDetails.vue (hunk 3 above)
Comment (holgerkoser) [reply frontend/src/components/NewShoot/NewShootDetails.vue:257]: Must be awaited

Comment (holgerkoser) [reply frontend/src/components/NewShoot/NewShootDetails.vue:257]: ✔️

File: frontend/src/components/ShootHibernation/HibernationConfiguration.vue (hunk 4 above)
Comment (holgerkoser) [reply frontend/src/components/ShootHibernation/HibernationConfiguration.vue]: ✔️

**Review by grolu — APPROVED** [review]
/lgtm

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**